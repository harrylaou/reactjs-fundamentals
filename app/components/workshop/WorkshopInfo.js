import React, {Component} from 'react'
import {
  Card, CardActions, CardHeader, CardTitle
} from 'material-ui/Card'
import FlatButton from 'material-ui/FlatButton'

const WorkshopInfo = (props) => {
  let workshop = props.workshop
  let workshopID = props.workshopID
  let instructors = ''
  let title = ''
  let button = ''

  if (workshop && workshop.id != workshopID) {
    props.fetchWorkshop(workshopID)
  }

  if (workshop) {
    instructors = workshop.instructors.map((instructor, index) => (
      <CardHeader
        key={index}
        title={instructor.name}
        subtitle={instructor.url}
        avatar={instructor.avatar}
      />
    ))
    title = `${workshop.title}  -  ${workshop.price}`
    button = (<FlatButton label="Buy" href={workshop.url} />)
  }

  return (
    <Card>
      <CardTitle title={title} />
      <CardTitle title="Instructors" />
      {instructors}
      <CardActions>
        {button}
      </CardActions>
    </Card>
  )
}

WorkshopInfo.propTypes = {
  workshop: React.PropTypes.object,
  workshopID: React.PropTypes.string
}

export default WorkshopInfo
