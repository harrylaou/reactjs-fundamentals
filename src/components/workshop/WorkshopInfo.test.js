import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import WorkshopInfo from './WorkshopInfo'

describe('<WorkshopInfo />', () => {
  it('should render with default props', () => {

    const wrapper = shallow(
      <WorkshopInfo />
    )
    expect(shallowToJson(wrapper.dive())).toMatchSnapshot()
  })
})