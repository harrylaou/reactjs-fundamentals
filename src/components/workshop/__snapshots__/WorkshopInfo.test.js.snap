// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkshopInfo /> should render with default props 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "zIndex": 1,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <CardTitle
      title=""
    />
    <CardTitle
      title="Instructors"
    />
    <CardActions />
  </div>
</Paper>
`;
